platforms/core-runtime/base-services/src/main/java/org/gradle/internal/operations/BuildOperationExecutor.java

```diff
@@ -36,15 +37,14 @@
  */
 @ThreadSafe
 @ServiceScope(Scopes.BuildSession.class)
-public interface BuildOperationExecutor extends BuildOperationRunner {
+public interface BuildOperationExecutor {
```

Comment:
💭 This is where we separate `BuildOperationExecutor` more explicitly from `BuildOperationRunner`.